source/jormungandr/tests/departure_board_tests.py

```diff
@@ -1104,3 +1103,107 @@ def test_routes_schedule_with_invalid_forbidden_uri(self):
         _, code = self.query_region("routes/line:A:0/route_schedules?"
                                      "from_datetime=20120615T080000&forbidden_uris[]=toto", check=False)
         assert code == 404
+
+
+@dataset({"timezone_cape_verde_test": {}})
+class TestFirstLastDatetime(AbstractTestFixture):
+    def test_first_last_datetime(self):
+        """
+        test first/last datetime
+
+        * The opening time should be the earliest departure of all vjs of the line
+        * The closing time should be the latest arrival of all vjs of the line
+
+        The closing time of line X is 02:45(local time) and the opening time is 8:00(local time)
+
+        Datetimes shown in the figure are all local datetimes
+         """
+
+        if not app.config['USE_SERPY']:
+            return
+
+        """              20170102
+                               from_datetime 20170102T1630       
+                                    |  
+                                    v
+         (Open)                             (Close)
+        08:00             15:00             26:45  
+         X_s1 ------------ X_S2 ------------ X_S3
+         
+        """
+        response = self.query_region("stop_points/X_S2/stop_schedules?from_datetime=20170102T1630"
+                                     "&data_freshness=base_schedule")
+
+        stop_schedules = response['stop_schedules']
+        assert stop_schedules
+        assert stop_schedules[0]['first_datetime']['date_time'] == "20171002T080000"
```

Comment:
shouldn't it be 16:00 and 18:00 one way and same the other way, as we are at X_S2 ?